src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitor.scala

```diff
@@ -147,7 +147,13 @@ class SubmissionMonitor(workspaceName: WorkspaceName,
         )
 
         if (refreshedSubmission.workflows.forall(workflow => workflow.status.isDone)) {
-          containerDAO.submissionDAO.update(workspaceContext, refreshedSubmission.copy(status = SubmissionStatuses.Done), txn)
+          val originalStatus = containerDAO.submissionDAO.get(workspaceContext, refreshedSubmission.submissionId, txn).get.status
+          originalStatus match {
+            case SubmissionStatuses.Submitted =>
+              containerDAO.submissionDAO.update(workspaceContext, refreshedSubmission.copy(status = SubmissionStatuses.Done), txn)
+            case SubmissionStatuses.Aborting =>
+              containerDAO.submissionDAO.update(workspaceContext, refreshedSubmission.copy(status = SubmissionStatuses.Aborted), txn)
+          }
```

Comment:
It should only ever be Submitted or Aborting in this stage, but it probably should be more robust than this. Will address.